### ISSUE TITLE
intl.locale.requested with GPO

### ISSUE THREAD

**Issue by lohitashwa2020** [mozilla/policy-templates#536]
I am installing Firefox 68.4 English installer with multiple language packs. My goal is to manage all configuration with GPO and not use Autoconfig at all.
I want firefox to follow system language, I can get this working well with Autoconfig file very well but not with GPO.
My understanding is that I need to set "Firefox/Requested locale" to empty and that should work. I can't get this to work by putting '''' 

Am I missing something? Sorry if this is something obvious or stupid.

**Comment by mkaply** [comment]
That's definitely how it should work.

Note there are two GPOs (one was deprecated).

Just enable Requested locale (String) and leave it blank and you should get the behavior you want.

**Comment by lohitashwa2020** [comment]
Thanks for getting back. I have enabled Requested locale (String) in Computer Configuration(Machine Policies). I can see in rsop.msc that GPO is enabled.
But If I check value of intl.locale.requested on browser it shows "und" and language switch doesn't work if I change system language.

**Comment by lohitashwa2020** [comment]
Futher update on my tests. intl.locale.requested ends up as "und" (for undefined I guess) when I leave Requested locale (String) to blank either in Computer Configuration or User Configuration.
Please note that I can get any language pack to work if I put values like Ja,Fr which means my GPOs are getting applied on machine/user.

**Comment by lohitashwa2020** [comment]
Some screenshots for reference-
GPO Configured:
![image](https://user-images.githubusercontent.com/61310511/75230626-fc432500-57d9-11ea-8096-7e660c59969e.png)
GPO Applied on machine:
![image](https://user-images.githubusercontent.com/61310511/75230666-17159980-57da-11ea-9442-1b4d4a19f2f8.png)
intl.locale.requested ends up as "und" :
![image](https://user-images.githubusercontent.com/61310511/75230708-2b599680-57da-11ea-9297-ef2a44a73961.png)

**Comment by mkaply** [comment]
This value is not being set by our code from what I can tell. It appears to be a user set value that you need to clear.

The policy does not change the pref intl.locale.requested.

Does this happen with a new profile?

**Comment by lohitashwa2020** [comment]
> This value is not being set by our code from what I can tell. It appears to be a user set value that you need to clear.
> I am 100% sure that "intl.locale.requested" is getting populated with "und" ONLY when I leave Requested locale (String) blank.

> The policy does not change the pref intl.locale.requested.
> If I put "Fr" in "Requested locale (String)", I can see "intl.locale.requested" populated with "Fr", same works if I put Multiple values like "Fr,Ja" etc. 

> Does this happen with a new profile?
I have tested this with freshly built machines as well. 

This one specific issue is holding back a large scale deployment of Firefox 68 ESR at my organization(45k machines). It would be great if I can get this working without using autoconfig files.

**Comment by lohitashwa2020** [comment]
I can not figure out what am I doing incorrectly. Whether I am creating Computer configuration or User Configuration Policy, "intl.locale.requested" always ends up undefined "und" when I enable "Requested Locale (String)" setting. 
I can see in registry, value is created correctly. 
![image](https://user-images.githubusercontent.com/61310511/75774267-a2140800-5d75-11ea-9e56-8e9e18a90c00.png)

But in Browser, "intl.locale.requested" ends up as "und" whether I have language packs installed or not.
![image](https://user-images.githubusercontent.com/61310511/75774341-ca036b80-5d75-11ea-9d84-3c9aafa104e9.png)
![image](https://user-images.githubusercontent.com/61310511/75774350-ce2f8900-5d75-11ea-9a12-69fa47dcb2f5.png)

**Comment by mkaply** [comment]
If you go to about:config, search on intl.locale.requested, right click and select "reset" and then restart Firefox, does it come back?

**Comment by mkaply** [comment]
I was finally able to recreate this. Debugging now.

**Comment by mkaply** [comment]
Patch is here:

https://bugzilla.mozilla.org/show_bug.cgi?id=1617455

trying to get it into the next Firefox.

**Comment by lohitashwa2020** [comment]
> If you go to about:config, search on intl.locale.requested, right click and select "reset" and then restart Firefox, does it come back?

Yes, even if I reset "intl.locale.requested", "und" comes back after next GPUPDATE.

**Comment by lohitashwa2020** [comment]
> Patch is here:
> 
> https://bugzilla.mozilla.org/show_bug.cgi?id=1617455
> 
> trying to get it into the next Firefox.

Good news. I look forward for this as this has been a blocking point for ESR deployment at my company.

**Comment by mkaply** [comment]
This fix made into the ESR that was released today. Can you give it a try?

**Comment by lohitashwa2020** [comment]
> This fix made into the ESR that was released today. Can you give it a try?

I have tested this and "und" is finally gone.
Thanks a lot for getting this fixed.

**Comment by mkaply** [comment]
Thanks for helping me figure this out!
